### PR TITLE
Fix plotting issues and debug `manage_agg_power`

### PR DESCRIPTION
**Pull request by julie-vienne**
This PR will take care of small issues that do not add or change features.

### CODE REVIEW

File: src/post_analysis/plotting/plot_episode_results.py [hunk 1]
```diff
@@ -602,7 +602,7 @@ def plot_imp_exp_violations(
             ax1.axhline(y=-prm['grd']['max_grid_export'], color='k', linestyle='dotted')
             ax1.set_ylabel('Grid import/export [kWh]')
             ax2.set_ylabel('System penalty [£]')
-            ax2.set_ylim([0, 0.11])
+            ax2.set_ylim([0, 1.1 * max(import_export_costs)])
```
Comment (floracharbo) [inline src/post_analysis/plotting/plot_episode_results.py:605]: is this not done automatically if you remove `set_ylim`?

File: src/post_analysis/plotting/plot_moving_average_rewards.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def plot_results_all_repeats(prm, record, moving_average=True, diff_to_opt=False
     if moving_average:
         plt.title('Moving average of difference between baseline and reward')
 
-    plt.gca().set_yticks(np.arange(-0.15, 0.2, 0.05))
+    #plt.gca().set_yticks(np.arange(-0.15, 0.2, 0.05))
```
Comment (floracharbo) [inline src/post_analysis/plotting/plot_moving_average_rewards.py:68]: remove if not used? or would you like to keep it to sometimes use it?

**Review by floracharbo — CHANGES_REQUESTED** [review]

**Comment by julie-vienne** [comment]
@floracharbo I've made some changes to debug running only `manage_agg_power`. Let me know what you think and if we can merge with main as I need it to run some experiments.

File: src/simulations/local_elec.py [hunk 3]
```diff
@@ -54,7 +54,7 @@ def __init__(self, prm):
         self.n_homes = prm['syst']['n_homes']
         self.homes = range(self.n_homes)
 
-        if self.prm['grd']['manage_voltage']:
+        if self.prm['grd']['manage_voltage'] or self.prm['grd']['manage_agg_power']:
```
Comment (floracharbo) [inline src/simulations/local_elec.py]: no let's not create the whole network object just to manage agg power. This is unnecessary.

File: src/simulations/local_elec.py [hunk 4]
```diff
@@ -348,6 +348,10 @@ def step(
             if record:
                 loads_flex = np.zeros(self.n_homes) if self.date == self.date_end \
                     else [sum(batch_flex[home][h][1:]) for home in homes]
+                loaded_buses = self.network.loaded_buses if self.prm['grd']['manage_voltage'] \
+                    else None
+                sgen_buses = self.network.sgen_buses if self.prm['grd']['manage_voltage'] \
+                    else None
```
Comment (floracharbo) [inline src/simulations/local_elec.py]: yes! I noticed this problem also. My bad I caused this problem when I debugged the comparison between optimisation and pandapower. I had fixed this in the main branch in parallel.

File: src/simulations/optimisation.py [hunk 5]
```diff
@@ -50,6 +50,8 @@ def res_post_processing(self, res):
             )
             res['hourly_line_losses'] = res['hourly_line_losses_pu'] * self.grd['base_power'] / 1000
         else:
+            res["voltage_squared"] = np.zeros([1, self.N])
+            res['hourly_line_losses'] = np.zeros(self.N)
```
Comment (floracharbo) [inline src/simulations/optimisation.py]: yes thanks for this. Sorry to have caused this issue. I think we duplicate the work as I had fixed this in the main branch.

File: main.py [hunk 6]
```diff
@@ -37,13 +38,14 @@
...
         'penalty_overvoltage': 0.1,
         'penalty_undervoltage': 0.1,
         'max_voltage': 1.001,
         'min_voltage': 0.999,
         'weight_network_costs': 1,
-        'subset_line_losses_modelled': 30
+        'subset_line_losses_modelled': 1,
+        'compare_pandapower_optimisation': False
     }
 }
```
Comment (floracharbo) [inline main.py]: maybe it is best if we don't commit the main files if only the settings change? then it disturbs the settings of whatever experiment we were running. I will create an issue to have the settings be a separate file that we add on .gitignore.

File: src/simulations/optimisation.py (hunk 5 above)
Comment (floracharbo) [reply src/simulations/optimisation.py]: could you please merge the main branch with this branch to harmonise before merging else there will be conflicts as we have solved the same problem and edited the same lines.

**Review by floracharbo — CHANGES_REQUESTED** [review]

**Comment by julie-vienne** [comment]
@floracharbo this PR does not fix all a lot of problems but can still be merge easily. I believe it is now ready to be merged.

**Comment by floracharbo** [comment]
should I close this @julie-vienne ?

**Comment by floracharbo** [comment]
> @floracharbo this PR does not fix all a lot of problems but can still be merge easily. I believe it is now ready to be merged.

everything in this PR is also in https://github.com/floracharbo/MARL_local_electricity/pull/13. I think if we merge them both we will end up having conflicts because editing twice the same lines. We might as well just do the "computational burden" PR.

**Comment by julie-vienne** [comment]
Close, see #13

**Pull request closed without merging**